ui/src/components/Inventory/InventoryCreateModal.js

```diff
@@ -0,0 +1,127 @@
+import { Form, Formik } from 'formik'
+import NumberTextField from '../Form/NumberTextField'
+import TextField from '../Form/TextField'
+
+class InventoryCreateModal extends React.Component {
+  render() {
+    const {
+      formName,
+      handleDialog,
+      handleProduct,
+      title,
+      initialValues,
+
+      name,
+      productType,
+      description,
+      averagePrice,
+      amount,
+      unitOfMeasurement,
+      bestBeforeDate,
+      neverExpires
+    } = this.props
+    return (
+      <Dialog
+        open={this.props.isDialogOpen}
+        maxWidth='sm'
+        fullWidth={true}
+        onClose={() => { handleDialog(false) }}
+      >
+        <Formik
+          initialValues={initialValues}
+          onSubmit={values => {
+            handleProduct(values)
+            handleDialog(true)
+          }}>
+          {helpers =>
+            <Form
+              noValidate
+              autoComplete='off'
+              id={formName}
+            >
+              <DialogTitle id='alert-dialog-title'>
+                {`${title} Inventory`}
+              </DialogTitle>
+              <DialogContent>
+                <Grid container>
+                  <Grid item xs={12} sm={12}>
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='name'
+                      label='Name'
+                      component={TextField}
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='productType'
+                      label='Product Type'
+                      component={TextField} //Dropdown select that reselects the cufrently available products
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='description'
+                      label='Description'
+                      component={TextField}
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='averagePrice'
+                      label='Average Price'
+                      component={NumberTextField}
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='amount'
+                      label='Amount'
+                      component={NumberTextField}
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='unitOfMeasurement'
+                      label='Unit of Measurement'
+                      component={TextField} //Dropdown of constants
+                    />
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
```

Comment:
Since this is a date field, you'll want to make sure some sort of calendar appears. This can be done through native html and does not require an external dependency